### PR TITLE
CLEWS-27625 Improve BatchClient test coverage

### PR DESCRIPTION
**Pull request by itsjohnward**
Pre-req: https://github.com/gro-intelligence/api-client/pull/218

### CODE REVIEW

File: api/client/batch_client.py [hunk 1]
```diff
@@ -132,21 +132,13 @@ def log_request(start_time, retry_count, msg, status_code):
...
             for data_series in self._data_series_queue:
                 data_series['show_revisions'] = True
-        self.batch_async_queue(self.get_data_points, self._data_series_queue, None,
-                               self.add_points_to_df)
+        self.batch_async_get_data_points(self._data_series_queue,
+                                         output_list=self._data_frame,
+                                         map_result=self.add_points_to_df)
         return self._data_frame
 
     # TODO: deprecate  the following  two methods, standardize  on one
```
Comment (alexsherman) [inline api/client/batch_client.py:144]: If we are removing `get_data_points` and its usage in `get_df`, is this TODO comment about deprecating `batch_async_get_data_points` relevant anymore?

File: api/client/batch_client_test.py [hunk 2]
```diff
@@ -1,10 +1,184 @@
...
+        list_of_ranked_series_lists = self.client.batch_async_rank_series_by_source(
+            [mock_data_series, mock_data_series]
+        )
+        # There were two inputs, so there should be two outputs:
+        self.assertEqual(len(list_of_ranked_series_lists), 2)
+        for series_list in list_of_ranked_series_lists:
+            # Not necessarily true, but true given the mock_rank_series_by_source() function:
+            self.assertEqual(len(series_list), len(mock_data_series))
+            for series in series_list:
+                self.assertTrue('metric_id' in series)
```
Comment (alexsherman) [inline api/client/batch_client_test.py:183]: Why assert that metric is in series and not other entity types?

**Review by alexsherman — APPROVED** [review]
LGTM, just a few minor questions. Only tested on python 3

File: api/client/batch_client_test.py (hunk 2 above)
Comment (itsjohnward) [reply api/client/batch_client_test.py:183]: Could assert all of them, but I just picked metric_id randomly as a test set. Source is the important one, plus one "other"

File: api/client/batch_client.py (hunk 1 above)
Comment (itsjohnward) [reply api/client/batch_client.py:144]: Yeah I think that'll be taken care of when BatchClient and GroClient are merged together. I just want one `get_data_points()` function that does both sync and async versions, depending on what the input is and whether the environment supports async. User shouldn't have to care about the difference

**Pull request merged**